quality_check/11-rnaseq_bias.Rmd

```diff
@@ -0,0 +1,294 @@
+  # if there's more than one nonempty data.frame, pick the first one
+  # this is a hack but will do for this purpose
+  sample_info_df <- sample_annotations[[nonempty_index[1]]]$data
+  
+  # paired or single end read -- only paired end will have corrections for
+  # GC bias
+  library_layout <- "SINGLE"
+  gc_bias <- any(sample_info_df$gc_bias_correct)
+  if (gc_bias & !is.na(gc_bias))
+    library_layout <- "PAIRED"
+  
+  # the library types inferred by salmon quant
+  library_type <- unique(unlist(sample_info_df$library_types))
+  # if more than one library type was detected, just report it as multiple
+  # this will require extra examination
+  if (length(library_type) > 1)
+    library_type <- "multiple"
+
+  # what % of the reads for this sample are successfully mapped to the txome
+  # index?
+  percent_mapped <- extract_var_na_rm("percent_mapped") 
+  
+  # how many reads does this sample have?
+  num_reads <- extract_var_na_rm("num_processed")
+  
+  # compatible fragment ratio -- a measure of agreement with the inferred 
+  # library type
+  compatible_fragment_ratio <- extract_var_na_rm("compatible_fragment_ratio")
+  
+  # what length index was used? (short or long)
+  index_length <- extract_var_na_rm("index_length")
+  
+  data.frame(
+    accession_code,
+    experiment = project_accession,
+    library_layout,
+    platform = sample_info$results$pretty_platform,
+    num_reads,
+    percent_mapped,
+    library_type,
+    compatible_fragment_ratio,
+    index_length
+  )
+  
+}
+```
+
+```{r}
+get_metadata_df <- function(accession_codes) {
+  # given a vector of sample accession codes, return a data.frame with the
+  # information returned by extract_sample_metadata; we bind all the rows 
+  # together
+  # 
+  # Args:
+  #   accession_codes: a vector of sample accession codes (should be RNA-seq
+  #                    samples) to be used with extract_sample_metadata
+  #
+  # Returns:
+  #   metadata_df: the output of extract_sample_metdata (a 1 row data.frame for
+  #                each sample) for each sample, now with the rows in a single
+  #                data.frame via dplyr::bind_rows
+
+  metadata_list <- lapply(accession_codes, extract_sample_metadata)
+  # metadata_list is a list of 1 row data.frames, so there will be 
+  # length(metadata_list) warnings about coercing to vector
+  metadata_df <- suppressWarnings(dplyr::bind_rows(metadata_list))
+}
+```
+
+### Set seed for reproducibility
+
+```{r}
+set.seed(6345)
+```
+
+## Initial investigation of RNA-seq sample groups
+
+Read in the data for PC1 and PC2 from `05-pca_test_compendium` and filter to
+only the RNA-seq samples
+
+```{r}
+pc_file <- file.path("results", "larger_compendium_PC1and2.tsv")
+seq_df <- readr::read_tsv(pc_file) %>%
+  dplyr::filter(compendium_technology_labels != "MICROARRAY")
+```
+
+We're going to somewhat arbitrarily divide samples into two groups on the basis
+of their values for PC1.
+
+```{r}
+seq_df <- seq_df %>%
```

Comment:
Could just combine this pipe and the one above.

Reply:
I have a preference for breaking it up so that this chunk follows
> We're going to somewhat arbitrarily divide samples into two groups on the basis
of their values for PC1.